mythril/laser/ethereum/call.py

```diff
@@ -122,17 +122,32 @@ def get_call_data(global_state, memory_start, memory_size, pad=True):
     :return: Tuple containing: call_data array from memory or empty array if symbolic, type found
     """
     state = global_state.mstate
+    transaction_id = '{}_internalcall'.format(global_state.current_transaction.id)
     try:
         # TODO: This only allows for either fully concrete or fully symbolic calldata.
         # Improve management of memory and callata to support a mix between both types.
-        call_data = state.memory[util.get_concrete_int(memory_start):util.get_concrete_int(memory_start + memory_size)]
-        if len(call_data) < 32 and pad:
-            call_data += [0] * (32 - len(call_data))
+        calldata_from_mem = state.memory[util.get_concrete_int(memory_start):util.get_concrete_int(memory_start + memory_size)]
+        i = 0
+        starting_calldata = []
+        while i < len(calldata_from_mem):
+            elem = calldata_from_mem[i]
+            if isinstance(elem, int):
+                starting_calldata.append(elem)
```

Comment:
what if elem is bigger then (2^8) - 1

Reply:
That shouldn't be possible with a correct memory implemention, but the current memory system is not quite correct, so I don't know if it is possible or not.

If the int is bigger than 2^8-1, I don't know how we should handle it. If, for example there are prepended 0 bytes, it is impossible to know the correct size.

Reply:
what if you do BitVec(elem, 256) ?

Reply:
for documentation #563